content/en/network_performance_monitoring/devices/guide/build-ndm-profile.md

```diff
@@ -0,0 +1,149 @@
+further_reading:
+- link: "https://datadoghq.dev/integrations-core/tutorials/snmp/profile-format/"
+  tag: "Documentation"
+  text: "Profile format reference"
+- link: "https://datadoghq.dev/integrations-core/tutorials/snmp/sim-format/"
+  tag: "Documentation"
+  text: "Simulation data format reference"
+---
+
+Datadog Network Device Monitoring uses profiles for certain makes and models of network devices. This tutorial shows the steps for building a basic NDM profile that collects OID metrics from HP iLO4 devices.
+
+NDM profiles use SNMP concepts. For basic details on SNMP, refer to the [terminology][1].
+
+<div class="alert alert-warning">
+This guide is for advanced users. Most devices can be configured using the <a href="/network_performance_monitoring/devices/profiles#metric-definition-by-profile">Datadog profiles</a>.
+</div>
+
+## Research
+
+The first step to building a NDM profile is researching the device and determining the metrics to collect.
+
+### Device information
+
+Refer to the manufacturer's website or search the web to find the following information:
+
+- Device name, manufacturer, and [system object identifier][1].
+
+- Understand the device and its use case. Metrics vary between routers, switches, bridges, etc. For example, according to the [HP iLO wikipedia page][2], iLO4 devices are used by system administrators for remote management of embedded servers.
+
+- Available versions of the device, and the versions to target. For example, HP iLO devices exist in multiple versions. This tutorial is specifically targeting HP iLO4.
+
+- Supported MIBs, OIDs, and associated MIB files. For example, HP provides a MIB package for iLO devices [their website][3].
+
+**Note**: For more details on device use cases, see [Networking hardware][4].
+
+### Metrics selection
+
+Next, decide the metrics to collect. Devices often expose thousands of metrics and OIDs that can span dozens of MIBs.
+
+Some guidelines to help you in this process:
+
+- Keep the number of metrics between 10 and 40.
+- Explore base profiles to see which ones could be applicable to the device.
+- Explore manufacturer-specific MIB files looking for metrics such as:
+    - General health: status gauges
+    - Network traffic: bytes in/out, errors in/out
+    - CPU and memory usage
+    - Temperature: temperature sensors, thermal condition
+    - Power supply
+    - Storage
+    - Field-replaceable units ([FRU][5])
+
+## Implementation
+
+Datadog recommends following this development workflow:
+
+- Add a profile
+- Write tests
+- Use simulation data
+
+### Add a profile
+
+Add a profile by creating a `.yaml` file with the `sysobjectid` and metrics, for example:
+
+```yaml
+sysobjectid: 1.3.6.1.4.1.232.9.4.10
+
+metrics:
+  - MIB: CPQHLTH-MIB
+    symbol:
+      OID: 1.3.6.1.4.1.232.6.2.8.1.0
+      name: cpqHeSysUtilLifeTime
+```
+
+**Note**: `sysobjectid` can be a wildcard pattern to match a sub-tree of devices, for example: `1.3.6.1.131.12.4.*`.
+
+### Write tests
```

Comment:
To be able to run the tests, the user need to install `ddev` tooling:

https://datadoghq.dev/integrations-core/setup/#ddev

I think this is quite advanced and require user to have some knowledge how integrations works.

To be able to run the tests, they also need to clone `integrations-core` and add test in `snmp/tests/test_profile.py`. 

Actually, I'm not sure if we want to suggest `Write tests` since this is quite advanced and mostly intended for devs adding new profiles to `integrations-core`.

Reply:
+1 I think this is really internal usage, not so much intended for an external user to build using our testing framework. We should probably recommend testing on their own network devices instead.

Reply:
Removing this section in favor of testing with an actual device